nodedb.go

```diff
@@ -105,6 +107,15 @@ func newNodeDB(db dbm.DB, cacheSize int, opts *Options) *nodeDB {
 	}
 }
 
+// Adds the given into a set of keys accessed
+// Note: Used by Deep Subtrees to know which keys to add existence proofs for
+func (ndb *nodeDB) addTrace(key []byte) {
+	if ndb.keysAccessed == nil {
```

Comment:
Can't we just ensure that `ndb.keysAccessed != nil`? Initializing it in `newNodeDB` should be enough - or I'm missing something?

Reply:
We'd also need to add it back when resetting the keys which is in `SaveVersion` right now.